### PR TITLE
fix: instrument timing for slow requests

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- Added to help debug #7190
- to make it easier to measure how long operations take on the backend

## Changes

- Add a class to measure how long a group of code takes and output to logs
- Add request info to the response log. This makes it easier to track what endpoint cause an error.

## Testing steps

- Verified timing logs are printed.
- verified the request info was in the response log.

### CODE REVIEW

File: backend/common/server/config.py [hunk 1]
```diff
@@ -92,6 +92,7 @@ def after_request(response: Response):
                     status_code=response.status_code,
                     content_length=response.content_length,
                     response_time=time.time() - g.start,
+                    request=dict(url=request.path, method=request.method),
```
Comment (Bento007) [inline backend/common/server/config.py:95]:
```suggestion
```

**Review by nayib-jose-gloria — APPROVED** [review]
we should look into using datadog tracing for this long-term, but short-term this seems reasonable

**Comment by Bento007** [comment]
I wonder if we can do this already with datadog

**Comment by Bento007** [comment]
It looks like [it can](https://app.datadoghq.com/apm/traces?query=env:dev%20AND%20service:backend%20AND%20-@http.status_code:*%20operation_name:flask.request%20status:error&agg_m=count&agg_m_source=base&agg_t=count&cols=core_service,core_resource_name,log_duration,log_http.method,log_http.status_code&fromUser=false&graphType=flamegraph&historicalData=true&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanID=12282518231279283759&spanType=all&timeHint=1720457629077&trace=AgAAAZCTRA2VZ59ulAAAAAAAAAAYAAAAAEFaQ1RSQzRPQUFEVDFvaC13R05xd3VtUAAAACQAAAAAMDE5MDkzNWEtOGM5NC00YmExLWI3MzgtYTQ0N2E2Mzk4MmJm&traceID=668c1997000000008fe990a045b3ef82&traceQuery=&view=spans&start=1720457580000&end=1720457640000&paused=true). 

I'd rather not add the overhead in this case.

**Comment by Bento007** [comment]
> I wonder if we can do this already with datadog

It looks like [it can](https://app.datadoghq.com/apm/traces?query=env:dev%20AND%20service:backend%20AND%20-@http.status_code:*%20operation_name:flask.request%20status:error&agg_m=count&agg_m_source=base&agg_t=count&cols=core_service,core_resource_name,log_duration,log_http.method,log_http.status_code&fromUser=false&graphType=flamegraph&historicalData=true&messageDisplay=inline&query_translation_version=v0&shouldShowLegend=true&sort=time&spanID=12282518231279283759&spanType=all&timeHint=1720457629077&trace=AgAAAZCTRA2VZ59ulAAAAAAAAAAYAAAAAEFaQ1RSQzRPQUFEVDFvaC13R05xd3VtUAAAACQAAAAAMDE5MDkzNWEtOGM5NC00YmExLWI3MzgtYTQ0N2E2Mzk4MmJm&traceID=668c1997000000008fe990a045b3ef82&traceQuery=&view=spans&start=1720457580000&end=1720457640000&paused=true). 

I'd rather not add the overhead in this case.

**Pull request closed without merging**